_posts/2015-02-06-Utilising-API-Blueprint-in-API-Clients.md

```diff
@@ -0,0 +1,119 @@
+
+This example is going to make use of one of our example blueprints, “Polls API”. I’ve pulled out an excerpt of the example blueprint and it looks something like the following. You can find the full version over in our [examples](https://github.com/apiaryio/api-blueprint/blob/kylef/polls-example/examples/Polls%20API.md).
+
+```markdown
+# Polls
+Polls is a simple web service that allows consumers to view polls and vote in them.
+
+## Group Question
+Resource related to questions in the API.
+
+## Question [/questions/{question_id}]
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+
+### View a question detail [GET]
+
+## Choice [/questions/{question_id}/choices/{choice_id}]
+
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+    + choice_id (number) ... ID of the Choice in form of an integer
+
+### Vote [PUT]
+
+This action allows you to vote on a question's choice.
+
+## Questions collection [/questions{?page}]
++ Parameters
+    + page (optional, number) ... The page of questions to return
+
+### List all questions [GET]
+### Create a new question [POST]
+```
+
+We can pull out any of the actions from our blueprint by the resource and action name in Swift:
+
+```swift
+if let transition = blueprint?.transition("Choice", action: "Vote") {
+}
+```
+
+This transition allows us to introspect it’s values at run-time, allowing us to retrieve protocol information such as the HTTP method and URI along with all the URI parameters.
+
+```swift
+println(transition.uri)
+println(transition.method)
+
+for (name, parameter) in transition.parameters {
+  println(name)
+  println("\(parameter.value) \(parameter.defaultValue)")
+}
+```
+
+We can use the elegant Swift HTTP networking library [Alamofire](https://github.com/Alamofire/Alamofire) along with some convenience methods we’ve built to perform HTTP requests from our transition directly.
+
+```swift
+let baseURL = NSURL(string: "http://polls.apiblueprint.org/")
+let poll = [
+  "title": "What's your favourite programming language?"
+]
+
+request(baseURL, transition, attributes:poll).responseJSON { (_, _, JSON, _) in
+  println(JSON)
+}
+```
+
+For the other action, to vote on a question by it's ID. We have a URI parameter
+`id` and `choice_id`. We can substitute this into the URI and make the request like so:
+
+```swift
+let params = [
+  "id": 1,
+  "choice_id": 2,
+]
+
+request(baseURL, transition, parameters:params).responseJSON { (_, _, JSON, _) in
+  println(JSON)
+}
+```
+
+Which will result in performing a PUT request to `/questions/1/choices/2`.
+
+If you’re interested in giving this a go, check out our [installation guide](https://github.com/the-hypermedia-project/representor-swift/wiki/API-Blueprint) on using API Blueprint in Swift.
+
+Of course, we’ve integrated this into our existing code examples. So right from Apiary you can see how it looks to perform a transition in Swift.
```

Comment:
Looks good I would just add some call to action to the end (such as let us know what do you think or something like that).